### DISCUSSION TITLE
C++ conversion

### DISCUSSION THREAD

**Discussion by gigablast** [Motion-Project/motion#1573] [General]
It seems that motion has to be converted to C++ to use opencv nowadays? (because opencv abandoned support for C?)

It would be quite useful to integrate opencv, and could be done by essentially turning motion into a C library
which is then called by some newly created C++ code. And then passing pointers to C functions, which such functions are defined in the C++ code, to the motion code in order to do opencv-related processing. I'm not even sure this is the issue, but thought it would be an easy way to C++-ify motion.

**Comment by genius3000** [comment]
I believe OpenCV is already supported in the C++ fork of Motion.
See: https://github.com/Motion-Project/motionplus

**Comment by Mr-Dave** [comment]
Matt is correct. Limited OpenCV processing is in [MotionPlus](https://github.com/Motion-Project/motionplus).  It is currently able to use haar cascades for secondary detection.  (None of the other methods from OpenCV have been implemented due to limited time.)  

Motion will not be converted to C++ and instead has generally gone into maintenance only mode.   All of my efforts and all the Motion requested enhancements will instead only be put into [MotionPlus](https://github.com/Motion-Project/motionplus) unless there is a very compelling reason.
